lock/common.go

```diff
@@ -0,0 +1,47 @@
+// Package lock provides implementations for cancellable and distributed locks.
+package lock
+
+import (
+	"time"
+
+	"gopkg.in/src-d/go-errors.v0"
+)
+
+var (
+	ErrCanceled      = errors.NewKind("context canceled")
+	ErrAlreadyClosed = errors.NewKind("already closed")
+)
+
+type SessionConfig struct {
+	TTL time.Duration
```

Comment:
I suppose that this is the session time to live, but what happens if this time is reached? an error is thrown? when, where?